### ISSUE TITLE
Android - Payments - Default payment is disappearing after closed and re-open the app.

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#10021]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Prerequisite: add VBA to the account
1. Launch the app
2. Log in with account
3. Go to Setting - Payments
4. Select Saving account
5. Tap Make default payment method
6. Put your password and Save
7. Close the app
8. Open the app again

## Expected Result:
Default payment is not disappearing after closed and re-open the app

## Actual Result:
Default payment is disappearing after closed and re-open the app.

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.85.2

**Reproducible in staging?:** Yes

**Reproducible in production?:** No

**Email or phone of affected tester (no customers):** applausetester+0901abb@applause.expensifail.com/ Feya86Katya

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/179871170-13c52f7f-9168-4c5c-a435-e73aab995835.mp4

https://user-images.githubusercontent.com/93399543/179903218-5604d5bb-3ac1-4a1e-aa31-1c05a5255bf4.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by marcaaron** [comment]
seems related to https://github.com/Expensify/App/pull/9828

@MariaHCD the front end code looks good here and the linked Web-E PR is on production.

Will try to look at logs and see if there's any obvious reason why this wouldn't work.

**Comment by marcaaron** [comment]
Seems to be throwing here:

https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%2272d79a1c5ff9305d-LAX%22)+AND+timestamp:[2022-07-19T23:32:50.671Z+TO+2022-07-20T01:32:50.671Z]&index=logs_expensify-017732

>Throwing exception with message: '401 No user with that partner/userID/password' from auth/lib/Login.cpp:320

**Comment by marcaaron** [comment]
@kbecciv are you certain that the password entered is correct? This looks like a normal error that would throw if the password was entered incorrectly and from watching the video I think that might have happened.

**Comment by kbecciv** [comment]
@marcaaron Yes, password is correct. Just re-test with build 1.1.85.5 and issue is reproduced. 
Also noticed the payments list is not displayed when navigate to Payments. 
Please attached video.

https://user-images.githubusercontent.com/93399543/180068986-5d632ec9-df0f-478f-888c-26d8aa3ba15c.mp4

**Comment by MariaHCD** [comment]
I tested with the same account and looks like there was an issue with setting the default payment method which is why the default badge doesn't stick.

```
2022-07-21 09:44:46 882 | applausetester+0901abb@applause.expensifail.com | Cannot find bankAccountID: 0

2022-07-21 09:44:46 882 | applausetester+0901abb@applause.expensifail.com | [OOPS!] API Threw: Auth SetWalletLinkedAccount returned an error ExpException - 4bbb72ecefead2582ca5df4a9b782afc ~~ message: '404 Bank account not found' 0: '[onyxMethod: 'merge' key: 'userWallet' value: '[error: 'Something went wrong. Please chat with Concierge for further assistance.']']' exceptionMessage: 'Auth SetWalletLinkedAccount returned an error' exceptionFile: '/git/releases/expensify.com/4c925fa/lib/Auth.php' exceptionLine: '117' exceptionCode: '404'
```

Logs: https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%2272e2fffc6d40b8fd-DEN%22)+AND+timestamp:[2022-07-21T08:44:46.877Z+TO+2022-07-21T10:44:46.877Z]&index=logs_expensify-017754

![Screen Shot 2022-07-21 at 1 45 52 PM](https://user-images.githubusercontent.com/12268372/180194662-b3a8ec3d-5336-43e1-90ad-081a741ca07f.png)

**Comment by marcaaron** [comment]
Ok. So after this conversation in Slack this is most likely happening because tester does not have an active wallet.

https://expensify.slack.com/archives/C03TQ48KC/p1658426458912309

Myself and Chirag both tested these flows with Gold wallets and it works ok. So, I think we need some better error handling. But this is not a deploy blocker.

**Comment by MariaHCD** [comment]
Thanks for the fix here! Agreed that we'll need better error handling here. In light of [this](https://expensify.slack.com/archives/C02HWMSMZEC/p1658356668529449) announcement, I'll also have to update the format of the error data sent to the client.

**Comment by MariaHCD** [comment]
Introducing better error handling here: https://github.com/Expensify/App/pull/10322

Closing in favor of: https://github.com/Expensify/Expensify/issues/216152
